libraries/lib-xml/XMLAttributeValueView.h

```diff
@@ -0,0 +1,174 @@
+   explicit XMLAttributeValueView(int value) noexcept;
+   //! Construct a view of type UnsignedInteger from the value.
+   explicit XMLAttributeValueView(unsigned int value) noexcept;
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(long value) noexcept;
+   //! Construct a view of type UnsignedInteger from the value.
+   explicit XMLAttributeValueView(unsigned long value) noexcept;
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(long long value) noexcept;
+   //! Construct a view of type UnsignedInteger from the value.
+   explicit XMLAttributeValueView(unsigned long long value) noexcept;
+   //! Construct a view of type Float from the value.
+   explicit XMLAttributeValueView(float value) noexcept;
+   //! Construct a view of type Double from the value.
+   explicit XMLAttributeValueView(double value) noexcept;
+   //! Construct a view of type StringView from the value.
+   explicit XMLAttributeValueView(const std::string_view& value) noexcept;
+
+   //! Get the view type.
+   Type GetType() const noexcept;
+
+   //! Check if view is Null.
+   bool IsNull() const noexcept;
+
+   //! Check if view has the SignedInteger type.
+   bool IsSignedInteger() const noexcept;
+   //! Check if view has the UnsignedInteger type.
+   bool IsUnsignedInteger() const noexcept;
+   //! Check if view has the Float type.
+   bool IsFloat() const noexcept;
+   //! Check if view has the Double type.
+   bool IsDouble() const noexcept;
+   //! Check if view has the StringView type.
+   bool IsStringView() const noexcept;
+
+   //! Try to get a boolean value from the view.
+   bool Get(bool& value) const noexcept;
+   //! Try to get a short value from the view.
+   bool Get(short& value) const noexcept;
+   //! Try to get an unsigned short value from the view.
+   bool Get(unsigned short& value) const noexcept;
+   //! Try to get an int value from the view.
+   bool Get(int& value) const noexcept;
+   //! Try to get an unsigned int value from the view.
+   bool Get(unsigned int& value) const noexcept;
+   //! Try to get a long value from the view.
+   bool Get(long& value) const noexcept;
+   //! Try to get an unsigned long value from the view.
+   bool Get(unsigned long& value) const noexcept;
+   //! Try to get a long long value from the view.
+   bool Get(long long& value) const noexcept;
+   //! Try to get an unsigned long long value from the view.
+   bool Get(unsigned long long& value) const noexcept;
+   //! Try to get a float value from the view.
+   bool Get(float& value) const noexcept;
+   //! Try to get a double value from the view.
+   bool Get(double& value) const noexcept;
+   //! Try to get a string_view value from the view.
+   bool Get(std::string_view& value) const noexcept;
+
+   //! Try to get boolean value or return defaultValue on failure.
+   bool GetBool(bool defaultValue = {}) const noexcept;
```

Comment:
You just overloaded one function name "Get" for the functions without default values.  Why not do thew same for this other series?

Overloading one name makes nice uses of Visit of a variant possible, with a simple generic lambda, as I did in this commit in master.  3acbcb4c2d8ada1e74991be79ecb3844568bcc0a

Reply:
I'm not sure if I understand what you mean. Get(bool&) and GetBoolean(bool) behavior is completely different.

Reply:
Instead of naming the overloads taking a default value as GetBool, GetInt16, ... they might all be called Get, but then a non-deduced template parameter would be needed when the default is not specified, thus Get<bool>().  That might be an extra bother to type, but if for some reason you wanted these functions in a variant visitor, there would be a uniform syntax Get<T>() for some T.

Well never mind.  I'm now reading how these functions are used in practice in a later commit.

Reply:
I can rename Get to `TryGet` if you want :-) This will allow to name this set of functions as simply `Get` then.

Reply:
I think you could just call them all Get and rely on overload resolution, but maybe you would not want to.

Or maybe Get_Or?  (See std::optional::value_or)

Reply:
>Get and rely on overload resolution

Overloading with reference and value type doesn't really play well.

Reply:
Reference and const reference is what I should have said